src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml

```diff
@@ -259,6 +291,65 @@ components:
         - targetDateRange
         - sequenceNumber
 
+    UpdateGoalRequest:
+      title: UpdateGoalRequest
+      type: object
+      description: A request to update an individual Goal that a Prisoner aims to complete as part of their Action Plan.
+      properties:
+        goalReference:
+          type: string
+          format: uuid
+          description: The Goal's unique reference
+          example: c88a6c48-97e2-4c04-93b5-98619966447b
+        title:
+          type: string
+          description: A title explaining the aim of the goal.
+          example: Improve communication skills
+        reviewDate:
+          type: string
+          format: date
+          description: An optional ISO-8601 date representing when the Goal is up for review.
+          example: '2023-12-19'
+        steps:
+          type: array
+          minItems: 1
+          description: A List of at least one Step.
+          items:
+            $ref: '#/components/schemas/UpdateStepRequest'
+        notes:
+          type: string
+          description: Some additional notes related to the Goal.
+          example: Pay close attention to Peter's behaviour.
+      required:
+        - goalReference
+        - title
+        - steps
+    UpdateStepRequest:
+      title: UpdateStepRequest
+      type: object
+      description: A request to create a Step within a Goal.
+      properties:
+        stepReference:
```

Comment:
LGTM 👍 
My only thought/question would be whether `stepReference` should be a mandatory field or not ... 🤔 
I'm thinking about a future story as part of "editing a goal", where the screens allow for adding steps as part of "editing a goal". In this case we wont have a `stepReference` (its a new step); so I'm wondering if the API needs to support it in this way and therefore the implementation will need to work out what is new/what is edited/what is deleted as part of working out the diff

What do you think?

Reply:
That's a good point, I'll change it 👍